components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -294,7 +297,14 @@ protected boolean handlePostAuthentication(Event event, String userName, UserSto
             Map<String, String> newClaims = new HashMap<>();
             newClaims.put(AccountConstants.FAILED_LOGIN_ATTEMPTS_CLAIM, currentFailedAttempts + "");
 
-            if (currentFailedAttempts >= maximumFailedAttempts) {
+            if (isAccountLockBypassRoleExist(userStoreManager, userName)) {
+                IdentityErrorMsgContext customErrorMessageContext = new IdentityErrorMsgContext(UserCoreConstants.ErrorCode.INVALID_CREDENTIAL,
+                        currentFailedAttempts, maximumFailedAttempts);
+                IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
+                String msg = String.format("Login attempt failed. Bypassing account locking for user %s", userName);
+                log.info(msg);
```

Comment:
Shall we add debug logs instead of info and audit logs.

Reply:
done